xmtp_mls/src/groups/sync.rs

```diff
@@ -213,16 +219,25 @@ where
                 let intent_data = SendMessageIntentData::from_bytes(intent.data.as_slice())?;
                 let group_id = openmls_group.group_id().as_slice();
                 let decrypted_message_data = intent_data.message.as_slice();
-                StoredGroupMessage {
-                    id: get_message_id(decrypted_message_data, group_id, envelope_timestamp_ns),
-                    group_id: group_id.to_vec(),
-                    decrypted_message_bytes: intent_data.message,
-                    sent_at_ns: envelope_timestamp_ns as i64,
-                    kind: GroupMessageKind::Application,
-                    sender_installation_id: self.client.installation_public_key(),
-                    sender_account_address: self.client.account_address(),
-                }
-                .store(conn)?;
+
+                let envelope = PlaintextEnvelope::decode(decrypted_message_data)
+                    .map_err(MessageProcessingError::DecodeError)?;
+
+                let (key, message) = if let Some(outer_content) = envelope.content {
+                    let Content::V1(V1 {
+                        idempotency_key,
+                        content,
+                    }) = outer_content;
+                    (idempotency_key, content)
+                } else {
+                    return Err(MessageProcessingError::InvalidPayload);
+                };
+
+                let message_id =
+                    calculate_message_id(group_id, &message, &self.client.account_address(), &key);
+
+                conn.set_delivery_status_to_published(&message_id)?;
+                conn.set_sent_at_ts(message_id, envelope_timestamp_ns)?;
```

Comment:
Could we merge these, so that there are fewer DB operations, and it's impossible to call one without the other, maybe by adding `delivery_timestamp_ns` as an argument to `set_delivery_status_to_published`?

Reply:
I collapsed these into one DB call that updates both fields 👍